torch/ao/quantization/quantizer/quantizer.py

```diff
@@ -157,6 +157,17 @@ class QuantizationAnnotation:
 
 
 class Quantizer(ABC):
+    # Allows for user defined transforms to run before annotating the graph.
+    # This allows quantizer to allow quantizing part of the model that are otherwise not quantizable.
+    # For example quantizer can
+    # a) decompose a compound operator like scaled dot product attention,
+    # into bmm and softmax if quantizer knows how to quantize bmm/softmax but not sdpa
+    # or b) transform scalars to tensor to allow quantizing scalares.
+    def transform_for_annotation(
```

Comment:
Since its a public method and not part of the `annotate`, one implication is that this is optional. Is it really optional?

Reply:
it is, we have the default implementation that returns model directly. you can see this is not an abstractmethod that need to implemented by all quantizers.

I can add some docs to clarify.

Reply:
I understand that for a custom Quantizer it is optional to extend it. That is not what I am talking about though.

I was referring to the call to this fn from `prepare_pt2d`, which is not optional. I didn't understand the rationale for having additional call vs. something annotate would do given it has to be before annotate and not optional.

I guess from function semantics pov it might make more sense to keep it seperate from annotate.

Reply:
yeah this is to keep the semantics for annotate clean, annotation should not mutate the graph, I'm also planning to change the return type for annotate to None as well in the future